src/pages/guides/using/idcs/idcs-events-data-stream-setup.md

```diff
@@ -0,0 +1,258 @@
+
+- Subscribe to the user driven change events of your choosing
+
+  ![Event selection](../../img/events_idcs_05.png "Select event subscriptions")
+
+- Set up OAuth Server-to-Server Credentials.
+  - The OAuth Server-to-Server credential relies on the OAuth 2.0 client_credentials grant type to generate access tokens.
+
+  ![Set up credentials](../../img/events_idcs_06.png "Set up credentials")
+
+- Set up Event Registration
+  - Provide a name and description for this event subscription
+
+  ![Set Up registration](../../img/events_idcs_07.png "Set Up registration")
+  
+- Configure Event Registration
+
+  ![Configure registration](../../img/events_idcs_08.png "Configure registration")
+
+  - Optionally choose whether to enable Webhook or Runtime action
+    - Enable Webhook
+      - We recommend batch over single webhooks
+      - For `Webhook URL` a public https endpoint must be provided
+      - The endpoint must be able to handle get and post requests
+      - The get request must respond with the challenge query if it exists
+      - The post request must respond that it received the message or the webhook will re-attempt to send several times before giving up and automatically disabling the webhook sends
+    - Enable Runtime action
+      - [See Setting up your Runtime Environment](/runtime/docs/guides/getting-started/)
+      - Select a pre-made runtime action/runtime namespace
+- After Saving
+
+  ![Verify setup](../../img/events_idcs_09.png "Verify setup")
+  
+  - Verify that the Status is `Active`
+  - If Webhook was selected, verify that it successfully passed the challenge without errors
+
+<DeveloperGuidelines/>
+
+### Event Data Structure
+
+*We are still finalizing the data structure as we prepare for Beta release. While we don't expect any major changes, there may be some minor modifications.*
+
+Events are structured in JSON format using the [CloudEvents](https://cloudevents.io/) spec
+
+*Example Event (batch)*
+
+````json
+
+{
+  "events": [
+    {
+      "position": "tuna:a1db5e65-7863-4110-bd45-2c4d2076871b.salmon:e1a7e7e0-39e2-453f-8c72-d70725020941.420c1092-a7fd-4132-9ee8-de7a41c01bcd.3.1720631450.o9y0pbhmarc-id2dxvat",
+      "event": {
+        "data": {
+          "key": "ec1b9eaf-a0b2-4bcf-9f6d-bd394e2586d7",
+          "source": "indesign",
+          "value": {
+            "jobId": "8b524c70-6779-47f1-ac2a-ce93c23b3f5f",
+            "eventId": "ec1b9eaf-a0b2-4bcf-9f6d-bd394e2586d7",
+            "state": "QUEUED",
+            "timestamp": 1720631447943
+          }
+        },
+        "id": "e414ef62-92f7-48f9-841b-b22d38f6141e",
+        "source": "urn:uuid:5c8f2d48-a2a7-4d0e-bb49-aabf9a72bd1e",
+        "specversion": "1.0",
+        "type": "QUEUED",
+        "datacontenttype": "application/json",
+        "time": "2024-07-10T17:10:48.985Z",
+        "event_id": "420c1092-a7fd-4132-9ee8-de7a41c01bcd",
+        "recipient_client_id": "7af7ab96045640b5a6e4724810145eb9"
```

Comment:
Do we have "recipient_client_id" sent in the current event schema?

Reply:
This was present in the response, left it for now.

Reply:
Yes, this field is added by the I/O Events team for digital signature verification. See [Improved and Resilient Security Verification for Webhook Events
](https://developer.adobe.com/events/docs/guides/#improved-and-resilient-security-verification-for-webhook-events)

Reply:
ok, thanks